### ISSUE TITLE
istioctl tries to exec into dead pilots

### ISSUE THREAD

**Issue by therealmitchconnors** [istio/istio#11488]
**Describe the bug**
The istioctl proxy-status command is supposed to exec pilot-discovery request from within the pilot pod, but when pilot has suffered evictions, it will target the evicted pod, rather than the live one.

**Expected behavior**
successful istioctl proxy-status output

**Actual behavior**
Error: error execing into istio-pilot-6d66894dd4-26tjr/istio-system discovery container: unable to upgrade connection: container not found ("discovery")

On further investigation, istio-pilot-6d66894dd4-26tjr is evicted.

**Steps to reproduce the bug**
Overload the cluster so that pilot gets evicted, then run istioctl proxy-status.

**Version**
```
version.BuildInfo{Version:"release-1.1-20190123-09-16", GitRevision:"2809bd07fde90e454bf7135e1d1737db57c38bc8", User:"root", Host:"a0d5b4f2-1eef-11e9-b805-0a580a2c0506", GolangVersion:"go1.10.4", DockerHub:"gcr.io/istio-release", BuildStatus:"Clean", GitTag:"1.1.0-snapshot.4-213-g2809bd0"}

Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.2", GitCommit:"cff46ab41ff0bb44d8584413b598ad8360ec1def", GitTreeState:"clean", BuildDate:"2019-01-10T23:35:51Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.11-gke.1", GitCommit:"5c4fddf874319c9825581cc9ab1d0f0cf51e1dc9", GitTreeState:"clean", BuildDate:"2018-11-30T16:18:58Z", GoVersion:"go1.9.3b4", Compiler:"gc", Platform:"linux/amd64"}
```

**Installation**
istio/tools/tree/master/perf/istio/setup.sh

**Environment**
GKE

**Comment by marcstreeter** [comment]
I'm using 1.0.5 and I turned off one of my pods (not thinking about it, since I assumed after a bit it would reschedule services to other pods).  After a couple days, I ran `istio proxy-status` and got the same error (`Error: error execing into istio-pilot-786dc4c88d-bw8pz discovery container: error dialing backend: dial tcp 192.168.1.203:10250: connect: no route to host`).  I then started up the pod I'd originally turned off and it works again.  So, yeah I'd expect it to just spin up another pilot service instead of waiting for the pod eternally.
